.github/workflows/build.yml

```diff
@@ -36,18 +37,23 @@ jobs:
             platform: windows-latest
             build-net-tokio: true
             build-no-std: true
+            build-futures: false
           - toolchain: beta
             build-net-tokio: true
             build-no-std: true
+            build-futures: true
           - toolchain: 1.41.1
             build-no-std: false
             test-log-variants: true
+            build-futures: false
```

Comment:
Now building on all platforms/toolchains with https://github.com/lightningdevkit/rust-lightning/pull/1744/commits/1a55192a1ebc6c65645e4e11c5dfcdec64f0bb8d.

Reply:
Yikes, this just got a bit more relevant, since it fails to build under MSRV. To quote the `futures` repo README:

"The current futures requires Rust 1.45 or later."

Reply:
Now 'solved' by pinning the `futures` version to the last one with MSRV 1.41 (https://github.com/lightningdevkit/rust-lightning/pull/1744/commits/6df23161ede65a67ec75d5397206ff1331aa0bbe). 

I'm however unsure why `macos-latest` and `windows-latest` targets are hanging with "Waiting for status to be reported". Any ideas?